src/app/shared/services/media-state.service.ts

```diff
@@ -26,42 +26,37 @@ export class MediaStateService implements OnDestroy {
   }
 
   private updateMedia(media: Media): void {
-    const mediaSubject: BehaviorSubject<Media | undefined> | undefined = this.mediaSubjects.get(media.sourceName)
-    if (mediaSubject) {
-      mediaSubject.next(media)
-      return
+    const mediaSubject: BehaviorSubject<Media | undefined> | undefined = this.getMediaSubject(media.sourceName)
```

Comment:
`this.getMediaSubject()` always returns a BehaviourSubject, so you can now write:
```
const mediaSubject: BehaviorSubject<Media | undefined>
```
instead of
```
const mediaSubject: BehaviorSubject<Media | undefined> | undefined
```

Reply:
Good catch!